Tests/SwiftParserTest/translated/UnclosedStringInterpolationTests.swift

```diff
@@ -26,99 +26,90 @@ final class UnclosedStringInterpolationTests: XCTestCase {
   func testUnclosedStringInterpolation2() {
     AssertParse(
       ##"""
-      _ = 1️⃣"mid == \(pete"
+      _ = "mid == \(pete1️⃣"2️⃣
       """##,
       diagnostics: [
-        // TODO: Old parser expected error on line 1: cannot find ')' to match opening '(' in string interpolation
-        // TODO: Old parser expected error on line 1: unterminated string literal
-        DiagnosticSpec(message: "expected expression"),
-        DiagnosticSpec(message: #"extraneous code '"mid == \(pete"' at top level"#),
+        DiagnosticSpec(locationMarker: "1️⃣", message: #"unexpected code '"' in string literal"#),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "expected ')' in string literal"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: #"expected '"' to end string literal"#),
```

Comment:
It'd be nice to handle this one better, ie. don't eat the `"` as unexpected.  That makes 7 worse, but IMO this would be by far the more common case.